File: flutter-libreria/backend/bussiness/shop/product.py
from libreria.backend.bussiness.shop.category import Category

class Product:
    def __init__(self, name: str, autor:str, editorial: str, collection: str, pages: int,  price: float, description: str, category: Category, stock: int, thumbnail: str, images: list[str]):
        self.id = None #TODO: gestionar IDs de esta y mas clases
        self.name = name
        self.price = price
        self.description = description
        self.category = category
        self.stock = stock
        self.thumbnail = thumbnail


class Book(Product):
    def __init__(self, name: str, autor:str, editorial: str, collection: str, pages: int,  price: float, description: str, category: Category, stock: int, thumbnail: str, images: list[str]):
        super().__init__(name, autor, editorial, collection, pages, price, description, category, stock, thumbnail)
        self.autor = autor
        self.editorial = editorial
        self.collection = collection
        self.pages = pages

File: flutter-libreria/frontend/libreria_front/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final String thumbnail;
  final double price;
  final List<String> images;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.thumbnail,
    required this.price,
    required this.images,
  });
}
